src/main.ts

```diff
@@ -42,23 +42,49 @@ app.get('/.well-known/server-health', (req, res) => {
   res.status(200).send('ok');
 });
 
-// The parameters we expect end users to provide
-interface BrazePocketHitsQuery {
-  scheduledSurfaceID: string;
+// The url params
+enum BrazePocketHitsParams {
+  scheduledSurfaceID = 'scheduledSurfaceID',
+}
+
+// The query parameters we expect end users to provide
+interface BrazePocketHitsQueryParams {
   date: string;
 }
 
-app.get('/scheduled-items/:scheduledSurfaceID?date=:date', async (req, res) => {
-  // enable 30 minute cache when in AWS
-  if (config.app.environment !== 'development') {
-    res.set('Cache-control', 'public, max-age=1800');
+// A custom request type to help with code completion
+type BrazePocketHitsRequest = Request<
+  Record<BrazePocketHitsParams, 'string'>,
+  any,
+  any,
+  BrazePocketHitsQueryParams
+>;
+
+// base path after our domain uri
+const basePath = '/scheduled-items/';
+
+/**
+ * Controller function for the path: /scheduled-items/:scheduledSurfaceID'
+ */
+app.get(
+  `${basePath}:${BrazePocketHitsParams.scheduledSurfaceID}`,
+  async (req: BrazePocketHitsRequest, res) => {
+    // This express app response Cache-control header is honoured by our CDN
+    // The cache max-age time limit used here is what the AWS CDN will use
+    if (config.app.environment !== 'development') {
+      res.set('Cache-control', 'public, max-age=1800');
+    }
+
+    //TODO: cache invalidation logic goes here? Confirm with stakeholders
+
+    // get our url param
+    const scheduledSurfaceID = req.params.scheduledSurfaceID;
+    // get our url query param
+    const date = req.query.date;
```

Comment:
Even though these are "typed" (no types at run time thanks TS 🙃 ) , we will still have to do validation to check if these values are `undefined` or not. The user could simply miss spell the param and it would be `undefined` here. I know Nina is doing that in her validation PR!